cpp/tests/io/json_test.cpp

```diff
@@ -1717,8 +1717,7 @@ TYPED_TEST(JsonFixedPointReaderTest, EmptyValues)
 
   cudf::io::json_reader_options const in_opts =
     cudf::io::json_reader_options::builder(
-      cudf::io::source_info{cudf::host_span<std::byte const>(
-        reinterpret_cast<std::byte const*>(buffer.data()), buffer.size())})
+      cudf::io::source_info{cudf::host_span<char const>(buffer)})
```

Comment:
wait, how does this work?

Reply:
This is the new API I just added that is templated. I'm playing around with how we can do this without as much refactoring on the user's end.
``` 
template <typename T, CUDF_ENABLE_IF(is_byte_like_type<T>())>
explicit source_info(std::vector<cudf::host_span<T const>> const& host_buffers)
```

I also have 
```
template <typename T, CUDF_ENABLE_IF(is_byte_like_type<T>())>
explicit source_info(T const* host_data, size_t size)
```

Which should make all the changes go away. I don't know if we want to hide the `host_span` behind the API though.